crates/cli/src/metaboard.rs

```diff
@@ -20,6 +22,60 @@ pub fn generate_emit_meta_calldata(meta: RainMetaDocumentV1Item) -> Result<Vec<u
     Ok(generate_emit_data_calldata(hash.into(), meta_bytes))
 }
 
+/// Deployment data for publishing to MetaBoard
+#[derive(Debug, Clone, Serialize, serde::Deserialize)]
+pub struct DeploymentData {
+    /// Keccak256 hash of the content (hex-encoded with 0x prefix)
+    pub subject: String,
+    /// CBOR-encoded metadata bytes (hex-encoded with 0x prefix)
+    pub meta_bytes: String,
+    /// Complete calldata for MetaBoard.emitMeta() (hex-encoded with 0x prefix)
+    pub calldata: String,
+}
+
+/// Validate dotrain content - basic checks
+fn validate_dotrain_content(content: &str) -> Result<(), Error> {
+    // Check if content is empty or only whitespace
+    if content.trim().is_empty() {
+        return Err(Error::InvalidInput(
+            "Dotrain content cannot be empty".to_string(),
+        ));
+    }
+
+    // Try to create DotrainSourceV1 to ensure it's valid
+    let _dotrain_source = DotrainSourceV1(content.to_string());
+
+    Ok(())
+}
+
+/// Generate deployment data for DotrainSourceV1 content
+/// Validates content and returns subject hash, meta bytes, and calldata
+pub fn generate_dotrain_deployment(content: &str) -> Result<DeploymentData, Error> {
+    // Validate content
+    validate_dotrain_content(content)?;
+
+    // Create DotrainSourceV1
+    let dotrain_source = DotrainSourceV1(content.to_string());
+
+    // Convert to RainMetaDocumentV1Item
+    let document: RainMetaDocumentV1Item = dotrain_source.into();
+
+    // Generate CBOR bytes
+    let meta_bytes = document.cbor_encode()?;
+
+    // Calculate subject hash
+    let subject_hash = document.hash(false)?;
+
+    // Generate calldata
+    let calldata = generate_emit_data_calldata(subject_hash.into(), meta_bytes.clone());
+
+    Ok(DeploymentData {
+        subject: format!("0x{}", hex::encode(subject_hash)),
```

Comment:
we can use `encode_prefixed` function to get a "0x" prefixed data. that way we don't have to use format here